### PR TITLE
[ML] fix(next-pylint): Resolve `no-name-in-module`, `non-abstract-transport-import`, `client-method-missing-tracing-decorator` and `networking-import-outside-azure-core-transport`

### PR DESCRIPTION
**Pull request by kdestin**
# Description

This pull request is part of a effort to ready the Azure ML Python SDK for an impending bump to the versions of pylint and azure-pylint-guidelines-checker that is used in our CI.

This pull request specifically resolves:

* [`E0611` (no-name-in-module)](https://pylint.pycqa.org/en/latest/user_guide/messages/error/no-name-in-module.html)
* `non-abstract-transport-import`, `client-method-missing-tracing-decorator` and `networking-import-outside-azure-core-transport` from the [azure-pylint-guidelines-checker](https://github.com/Azure/azure-sdk-tools/blob/39a5ef969b8d1a297100d53c1a13c3edc7ab3248/tools/pylint-extensions/azure-pylint-guidelines-checker/README.md#rules-list) checker.

Validated with:

`pylint==2.15.8`
`azure-pylint-guidelines-checker=0.0.8`

`pylint --rcfile=../../../eng/pylintc --disable=all --enable=E0611,non-abstract-transport-import,client-method-missing-tracing-decorator,networking-import-outside-azure-core-transport azure/`

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_artifact_utils.py [hunk 1]
```diff
@@ -15,8 +15,7 @@
 from pathlib import Path
 from threading import Lock
 
-import requests
```
Comment (diondrapeck) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_artifact_utils.py:18]: good catch - I didn't realize we still had use of requests lurking around

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/compute/compute_instance.py [hunk 2]
```diff
@@ -46,9 +46,9 @@ class OsImageMetadataSchema(PathAwareSchema):
 
     @post_load
     def make(self, data, **kwargs):
-        from azure.ai.ml.entities import OsImageMetadata
+        from azure.ai.ml.entities import ImageMetadata
```
[the remark this answers is not in the document]
Comment (kdestin) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/compute/compute_instance.py:49]: OsImageMetadata doesn't exist. I think it should've been ImageMetadata. @nthandeMS do you have any more context on this?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_artifact_utils.py [hunk 3]
```diff
@@ -169,15 +168,21 @@ def _redirect_artifacts_tool_path(self, organization):
             token = credential.get_token("https://management.azure.com/.default")
             header = {"Authorization": "Bearer " + token.token}
 
+            # The underlying HttpTransport is meant to be user configurable.
+            # MLClient instances have a user configured Pipeline for sending http requests
+            # TODO: Replace this with MlCLient._requests_pipeline
```
[the remark this answers is not in the document]
Comment (kdestin) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_artifact_utils.py:173]: Done

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/compute/compute_instance.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/compute/compute_instance.py:49]: Cool, good catch, Kevin!

**Review by diondrapeck — APPROVED** [review]

**Pull request merged**